beetsplug/deezer.py

```diff
@@ -192,21 +206,42 @@ def track_for_id(self, track_id=None, track_data=None):
             deezer_id = self._get_id("track", track_id, self.id_regex)
             if deezer_id is None:
                 return None
-            track_data = requests.get(self.track_url + deezer_id).json()
-            if "error" in track_data:
+            try:
+                track_data = requests.get(
+                    self.track_url + deezer_id, timeout=10
+                ).json()
+            except requests.exception as e:
                 self._log.debug(
-                    f"Error fetching track {track_id}: "
-                    f"{track_data['error']['message']}"
+                    "Error fetching track {}\n Error: {}", deezer_id, e
                 )
                 return None
         track = self._get_track(track_data)
 
         # Get album's tracks to set `track.index` (position on the entire
         # release) and `track.medium_total` (total number of tracks on
         # the track's disc).
-        album_tracks_data = requests.get(
-            self.album_url + str(track_data["album"]["id"]) + "/tracks"
-        ).json()["data"]
+        try:
+            album_tracks_data = requests.get(
+                self.album_url + str(track_data["album"]["id"]) + "/tracks",
+                timeout=10,
+            ).json()["data"]
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.RequestException,
+        ) as e:
+            self._log.debug(
+                "Error fetching album tracks for {} \n Error: {}",
+                track_data["album"]["id"],
+                e,
+            )
+            return None
+        else:
+            if "error" in album_tracks_data:
+                self._log.debug(
+                    "Deezer API error: {}",
+                    album_tracks_data["error"]["message"],
+                )
+                return None
```

Comment:
This handler code appears to be duplicated `track_for_id` and `album_for_id`. What do you think about introducing a helper function? It would work as a wrapper for `requests.get` with additional error handling. Then we only need to include all this error-reporting code once and invoke it from both methods.